tensorboard/plugins/hparams/tf_hparams_parallel_coords_plot/axes.ts

```diff
@@ -0,0 +1,547 @@
+      console.error("Scale type is null, but brushSelection isn't: ",
+                    brushSelection);
+      return new AlwaysPassingBrushFilter();
+    }
+    switch (this._scaleType) {
+      case ScaleType.LINEAR:
+      case ScaleType.LOG: { /* Fall Through */
+        const [lower, upper] =
+          tf.hparams.parallel_coords_plot.continuousScaleInverseImage(
+            this.yScale(), brushSelection[0], brushSelection[1]);
+        return new IntervalBrushFilter(lower,
+                                       upper,
+                                       /*lowerOpen=*/ false,
+                                       /*upperOpen=*/ false);
+      }
+      case ScaleType.QUANTILE: {
+        const [lower, upper] =
+          tf.hparams.parallel_coords_plot.quantileScaleInverseImage(
+            this.yScale(), brushSelection[0], brushSelection[1]);
+        return new IntervalBrushFilter(lower,
+                                       upper,
+                                       /*lowerOpen=*/ false,
+                                       /*upperOpen=*/ true);
+      }
+      case ScaleType.NON_NUMERIC:
+        return new SetBrushFilter(
+          tf.hparams.parallel_coords_plot.pointScaleInverseImage(
+            this.yScale(), brushSelection[0], brushSelection[1]));
+    }
+    console.error("Unknown scale type: ", this._scaleType);
+    return new AlwaysPassingBrushFilter();
+  }
+
+  private readonly _svgProps: SVGProperties;
+  private readonly _schema: tf.hparams.Schema;
+  private readonly _interactionManager: InteractionManager;
+  private readonly _colIndex: number;
+  private _isDisplayed: boolean;
+  private _yScale: any;  /* D3 scale */
+  private _scaleType: ScaleType | null;
+  private _brushSelection: d3.BrushSelection;
+  private _brushFilter: AxisBrushFilter;
+}
+
+/**
+ * Manages the collection of axes shown in the plot. Has methods that handle
+ * dragging an axis and contains the logic for re-ordering the axes 
+ * during dragging. 
+ */
+export class AxesCollection {
+  public constructor(svgProps: SVGProperties, schema: tf.hparams.Schema,
+                     interactionManager: InteractionManager) {
+    this._svgProps = svgProps;
+    this._schema = schema;
+    this._axes = this._createAxes(interactionManager);
+    this._stationaryAxesPositions = d3.scalePoint()
+      .range([1, this._svgProps.width - 1])
+      .padding(0.5);
+    this._draggedAxis = null;
+    this._svgProps.svgG.selectAll("g.axis-parent").remove();
+    this._parentsSel = this._svgProps.svgG.selectAll(".axis-parent");
+  }
+
+  /**
+   * Updates all axes based on the given 'options' (see the comments in 
+   * tf-hparams-scale-and-color-controls.html) and sessionGroups. SessionGroups
+   * are used to update the domain (and thus scale) of the axes. The 'options'
+   * object control which axes are visible.
+   */
+  public updateAxes(options: any, sessionGroups: any[]) {
+    console.assert(!this.isAxisDragging());
```

Comment:
Instead of throwing exception, I'd silently exit or say something more user friendly especially if it is going to litter in the console.